RealmSwift/Impl/SchemaDiscovery.swift

```diff
@@ -19,52 +19,76 @@
+    // The Realm property type associated with this type
     static var _rlmType: PropertyType { get }
     static var _rlmOptional: Bool { get }
+    // Does this type require @objc for legacy declarations? Not used for modern
+    // declarations as no types use @objc.
+    static var _rlmRequireObjc: Bool { get }
+
+    // Set any fields of the property applicable to this type other than type/optional.
+    // There are both static and non-static versions of this function because
+    // some times need data from an instance (e.g. LinkingObjects, where the
+    // source property name is runtime data and not part of the type), while
+    // wrappers like Optional need to be able to recur to the wrapped type
+    // without creating an instance of that.
     func _rlmPopulateProperty(_ prop: RLMProperty)
     static func _rlmPopulateProperty(_ prop: RLMProperty)
-    static var _rlmRequireObjc: Bool { get }
 }
 
-extension _RealmSchemaDiscoverable {
+internal protocol SchemaDiscoverable: _RealmSchemaDiscoverable {}
+extension SchemaDiscoverable {
     public static var _rlmOptional: Bool { false }
     public static var _rlmRequireObjc: Bool { true }
     public func _rlmPopulateProperty(_ prop: RLMProperty) { }
     public static func _rlmPopulateProperty(_ prop: RLMProperty) { }
 }
 
-// If the property is a storage property for a lazy Swift property, return
-// the base property name (e.g. `foo.storage` becomes `foo`). Otherwise, nil.
-private func baseName(forLazySwiftProperty name: String) -> String? {
-    // A Swift lazy var shows up as two separate children on the reflection tree:
-    // one named 'x', and another that is optional and is named "$__lazy_storage_$_propName"
-    if let storageRange = name.range(of: "$__lazy_storage_$_", options: [.anchored]) {
-        return String(name[storageRange.upperBound...])
-    }
-    return nil
-}
-
 internal extension RLMProperty {
     convenience init(name: String, value: _RealmSchemaDiscoverable) {
         let valueType = Swift.type(of: value)
         self.init()
         self.name = name
         self.type = valueType._rlmType
         self.optional = valueType._rlmOptional
-        valueType._rlmPopulateProperty(self)
         value._rlmPopulateProperty(self)
+        valueType._rlmPopulateProperty(self)
         if valueType._rlmRequireObjc {
             self.updateAccessors()
         }
     }
 }
 
-private func getProperties(_ cls: RLMObjectBase.Type) -> [RLMProperty] {
-    let object = cls.init()
+private func getModernProperties(_ object: ObjectBase) -> [RLMProperty] {
+    return Mirror(reflecting: object).children.compactMap { prop in
+        guard let label = prop.label else { return nil }
+        guard let value = prop.value as? DiscoverablePersistedProperty else {
+            return nil
+        }
+        let property = RLMProperty(name: label, value: value)
+        property.swiftIvar = ivar_getOffset(class_getInstanceVariable(type(of: object), label)!)
```

Comment:
Once we have the obj key and col key, why do we still need the ivar offset?

Reply:
To read from and write to unmanaged properties from obj-c.

Reply:
Also very importantly this is how we set the col key on the managed property wrappers.